### PR TITLE
Feature/password page

### PR DESCRIPTION
**Pull request by bittricky**
### What does this pull request aim to achieve? 💯

Implement the front end for issue #651 and provide a way for the user to recover their account.

_Please be concise and link any related issue(s):_
#651 

### 🚨Before submitting this pull request🚨:

_Please do **NOT** submit this PR if you have not done the following:_

1. I have checked that no one else is working on similar changes.
2. I have read the latest [**README**](https://github.com/FightPandemics/FightPandemics/blob/master/README.md) and understand the development workflow.
3. The name of this branch is: **`feature/<branch_name>`** (need to have cloned the repo not forked).
4. This branch is rebased/merged with the **latest master**.
5. There are no merge conflicts.
6. There are no console warnings and errors.
7. On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

File: client/src/pages/Login.js [hunk 1]
```diff
@@ -175,6 +206,7 @@ const Login = ({ isLoginForm }) => {
   );
   const [passwordType, setPasswordType] = useState("password");
   const [confirmPasswordType, setConfirmPasswordType] = useState("password");
+  const [sentRecoveryLink, setRecoveryLink] = useState(false);
```
Comment (joshmorel) [inline client/src/pages/Login.js:209]:
```suggestion
  const [recoveryLink, setRecoveryLink] = useState(false);
```

Seems to be the react naming convention

File: client/src/pages/Login.js [hunk 2]
```diff
@@ -278,117 +328,158 @@ const Login = ({ isLoginForm }) => {
...
+                  onClick={
+                    isLoginForm
+                      ? handleSubmit(onLoginWithEmail)
+                      : handleSubmit(onSignup)
+                  }
+                >
+                  {isLoginForm ? "Sign In" : "Sign Up"}
+                </SubmitButton>
+              </form>
+            ) : (sentRecoveryLink) ? (
```
Comment (joshmorel) [inline client/src/pages/Login.js:440]:
```suggestion
            ) : (recoveryLink) ? (
```

File: client/src/pages/Login.js (hunk 1 above)
Comment (bittricky) [reply client/src/pages/Login.js]: Pushed changes for the fix commit 6e7fefe.

File: client/src/pages/Login.js (hunk 2 above)
Comment (bittricky) [reply client/src/pages/Login.js]: Pushed changes for the fix commit 6e7fefe

**Comment by joshmorel** [comment]
@bittricky  It's still redirecting me to verify-your-email page the `forgotPassword` in `RouteWithSubRoute` is undefined, it would need to be passed from somewhere like `sessionReducer` via `mapStateToProps`. BUT leave this for now, I have an idea. The other issue #545 needs to be fixed and some of the same designs. Why don't we actually redirect to this page after forgot password then vary the content by passing some variable.

I'll merge this then start to work on #545 and we can address this issue as part of it.

**Pull request merged**